webapp/client/src/pages/TaxNumberPage.test.js

```diff
@@ -0,0 +1,402 @@
+        expect(screen.queryByText("Steuernummer")).not.toBeNull();
+        expect(screen.queryAllByRole("textbox")).toHaveLength(3);
+      });
+
+      it("Do not tax office selection", () => {
+        expect(
+          screen.queryByLabelText(
+            "Wählen Sie Ihr Finanzamt",
+            notExactQueryOptions
+          )
+        ).toBeNull();
+      });
+
+      it("Do not show confirmation field", () => {
+        expect(
+          screen.queryByLabelText("Hiermit bestätige", notExactQueryOptions)
+        ).toBeNull();
+      });
+
+      describe("Enter tax number and then toggle tax number available question", () => {
+        beforeEach(() => {
+          userEvent.type(screen.queryAllByRole("textbox")[0], "123");
+          fireEvent.click(screen.getByText("Nein"));
+          fireEvent.click(screen.getByText("Ja"));
+        });
+
+        it("Should show Bundesland with Baden-Württemberg selected", () => {
+          expect(
+            screen.queryByLabelText("Bundesland", notExactQueryOptions)
+          ).not.toBeNull();
+          expect(screen.queryAllByRole("combobox")[0]).toHaveValue("bw");
+        });
+
+        it("Tax number inputs are empty", () => {
+          expect(screen.queryAllByRole("textbox")[0]).not.toHaveValue();
+          expect(screen.queryAllByRole("textbox")[1]).not.toHaveValue();
+          expect(screen.queryAllByRole("textbox")[2]).not.toHaveValue();
+        });
+      });
+    });
+
+    describe("Select Hessen", () => {
+      beforeEach(() => {
+        userEvent.selectOptions(screen.getByRole("combobox"), ["he"]);
+      });
+
+      it("Should show not split tax number input", () => {
+        expect(screen.queryByText("Steuernummer")).not.toBeNull();
+        expect(screen.queryAllByRole("textbox")).toHaveLength(1);
+      });
+
+      it("Do not tax office selection", () => {
+        expect(
+          screen.queryByLabelText(
+            "Wählen Sie Ihr Finanzamt",
+            notExactQueryOptions
+          )
+        ).toBeNull();
+      });
+
+      it("Do not show confirmation field", () => {
+        expect(
+          screen.queryByLabelText("Hiermit bestätige", notExactQueryOptions)
+        ).toBeNull();
+      });
+
+      describe("Enter text and change state to Bayern", () => {
+        beforeEach(() => {
+          userEvent.type(screen.queryAllByRole("textbox")[0], "1234");
+          userEvent.selectOptions(screen.getByRole("combobox"), ["by"]);
+        });
+
+        it("Should show split tax number input", () => {
+          expect(screen.queryByText("Steuernummer")).not.toBeNull();
+          expect(screen.queryAllByRole("textbox")).toHaveLength(3);
+        });
+
+        it("Tax number inputs are empty", () => {
+          expect(screen.queryAllByRole("textbox")[0]).not.toHaveValue();
+          expect(screen.queryAllByRole("textbox")[1]).not.toHaveValue();
+          expect(screen.queryAllByRole("textbox")[2]).not.toHaveValue();
```

Comment:
Should we have a test that - if the value is previously saved in the cookie (set by the props) and I then change the Bundesland, the tax number stays intact?